cmd/allocator/main.go

```diff
@@ -193,18 +237,32 @@ func (h *httpHandler) allocateHandler(w http.ResponseWriter, r *http.Request) {
 	logger.WithField("request", request).Infof("allocation request received")
 
 	gsa := converters.ConvertAllocationRequestV1Alpha1ToGSAV1(&request)
-	allocation := h.agonesClient.AllocationV1().GameServerAllocations(gsa.ObjectMeta.Namespace)
-	allocatedGsa, err := allocation.Create(gsa)
+	resultObj, err := h.allocationCallback(gsa, h.stop)
 	if err != nil {
 		http.Error(w, err.Error(), httpCode(err))
 		logger.WithField("gsa", gsa).WithError(err).Info("calling allocation extension API failed")
 		return
 	}
 
+	w.Header().Set("Content-Type", "application/json")
+	if status, ok := resultObj.(*metav1.Status); ok {
+		w.WriteHeader(int(status.Code))
+		err = json.NewEncoder(w).Encode(status)
+		if err != nil {
+			http.Error(w, "internal server error", http.StatusInternalServerError)
+			logger.Error(err)
```

Comment:
Maybe add some context to the log, e.g. `logger.WithError(err).Error("Unable to encode status in json")`